File: launchschool_exercises/109_small_problems/easy5/daily_double.rb
def crunch(str)
  str.squeeze('0-9a-zA-Z')
end

puts crunch('ddaaiillyy ddoouubbllee') # == 'daily double'
puts crunch('4444abcabccba') # == '4abcabcba'
puts crunch('ggggggggggggggg') # == 'g'
puts crunch('a') # == 'a'
puts crunch('') # == ''

File: launchschool_exercises/210_small_problems/easy2/daily-double.js
function crunch(str) {
  var lastSeen = str[0];
  var shrunkStr = str[0];

  if (str.length === 0) return '';

  for (var i = 1; i < str.length; i++) {

    if (str[i] === lastSeen) {
      continue;
    } else {
      shrunkStr += str[i];
      lastSeen = str[i];
    }
  }

  return shrunkStr;
}

console.log(crunch('ddaaiillyy ddoouubbllee'));    // "daily double"
console.log(crunch('4444abcabccba'));              // "4abcabcba"
console.log(crunch('ggggggggggggggg'));            // "g"
console.log(crunch('a'));                          // "a"
console.log(crunch(''));                           // ""
